apps/web/src/ee/clerk/providers/useCreateAuthContextEnterprise.ts

```diff
@@ -0,0 +1,205 @@
+import { useEffect, useCallback, useState } from 'react';
+import { useLDClient } from 'launchdarkly-react-client-sdk';
+import { useLocation, useNavigate } from 'react-router-dom';
+import { useQueryClient } from '@tanstack/react-query';
+import { setUser as setSentryUser, configureScope } from '@sentry/react';
+import type { IOrganizationEntity, IUserEntity } from '@novu/shared';
+import { useAuth, useUser, useOrganization, useOrganizationList } from '@clerk/clerk-react';
+import { OrganizationResource, UserResource } from '@clerk/types';
+import { useSegment } from '../../../components/providers/SegmentProvider';
+import { PUBLIC_ROUTES_PREFIXES, ROUTES } from '../../../constants/routes';
+
+const toUserEntity = (clerkUser: UserResource): IUserEntity => ({
+  _id: clerkUser.id,
+  firstName: clerkUser.firstName,
+  lastName: clerkUser.lastName,
+  email: clerkUser.emailAddresses[0].emailAddress,
+  profilePicture: clerkUser.imageUrl,
+  createdAt: clerkUser.createdAt?.toString() ?? '',
+  showOnBoarding: clerkUser.publicMetadata.showOnBoarding,
+  showOnBoardingTour: clerkUser.publicMetadata.showOnBoardingTour,
+  servicesHashes: clerkUser.publicMetadata.servicesHashes,
+  jobTitle: clerkUser.publicMetadata.jobTitle,
+  hasPassword: clerkUser.passwordEnabled,
+});
+
+const toOrganizationEntity = (clerkOrganization: OrganizationResource): IOrganizationEntity => ({
+  _id: clerkOrganization.id,
+  name: clerkOrganization.name,
+  createdAt: clerkOrganization.createdAt.toString(),
+  updatedAt: clerkOrganization.updatedAt.toString(),
+  apiServiceLevel: clerkOrganization.publicMetadata.apiServiceLevel,
+  defaultLocale: clerkOrganization.publicMetadata.defaultLocale,
+  domain: clerkOrganization.publicMetadata.domain,
+  productUseCases: clerkOrganization.publicMetadata.productUseCases,
+});
+
+export function useCreateAuthContextEnterprise() {
+  const { signOut, orgId } = useAuth();
+  const { user: clerkUser, isLoaded: isUserLoaded } = useUser();
+  const { organization: clerkOrganization, isLoaded: isOrganizationLoaded } = useOrganization();
+  const { setActive, isLoaded: isOrgListLoaded } = useOrganizationList();
+
+  const [user, setUser] = useState<IUserEntity | undefined>(undefined);
+  const [organization, setOrganization] = useState<IOrganizationEntity | undefined>(undefined);
+
+  const ldClient = useLDClient();
+  const segment = useSegment();
+  const queryClient = useQueryClient();
+  const navigate = useNavigate();
+  const location = useLocation();
+
+  const inPublicRoute = Array.from(PUBLIC_ROUTES_PREFIXES.values()).find((prefix) =>
+    location.pathname.startsWith(prefix)
+  );
+  const inPrivateRoute = !inPublicRoute;
+
+  const logout = useCallback(() => {
+    queryClient.clear();
+    segment.reset();
+    navigate(ROUTES.AUTH_LOGIN);
+    signOut();
+  }, [navigate, queryClient, segment, signOut]);
+
+  const redirectTo = useCallback(
```

Comment:
This method shouldn't be needed in the Clerk realm as the components support redirect_url parameters. We can leave it for now but let's add a todo to get back to this.

Reply:
Yes I thought so as well, but It was added recently and I was not 100% sure so I kept it - we can remove this later when we confirm its not required.